libsolidity/codegen/ir/Common.h

```diff
@@ -22,11 +22,29 @@
 
 #include <libsolidity/ast/AST.h>
 
+#include <algorithm>
 #include <string>
 
 namespace solidity::frontend
 {
 
+/**
+ * Structure that describes arity and co-arity of a Yul function, i.e. the number of its inputs and outputs.
+ */
+struct YulArity
+{
+	explicit YulArity(size_t _in, size_t _out): in(_in), out(_out) {}
+
+	static YulArity fromDefinition(FunctionDefinition const& _function);
```

Comment:
Either you extend the description of the struct above or you shortly explain how you get from the Solidity function definiton to the arity of the yul function.

Reply:
OK. I'll add more info.

Reply:
Well, maybe it's better to remove it after all. There seem to be many ways to go from a definition to the type. I realized that I should probably include `internal` somewhere in the name and even then it would still be ambiguous. I could document it but it's probably better to have something like this in the code instead:
```c++
YulArity::fromType(*TypeProvider::function(*function, FunctionType::Kind::Internal))
```